google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerGaxRetryTest.java

```diff
@@ -130,7 +130,13 @@ public void setUp() throws Exception {
             .setChannelProvider(channelProvider)
             .setCredentials(NoCredentials.getInstance());
     // Make sure the session pool is empty by default.
-    builder.setSessionPoolOption(SessionPoolOptions.newBuilder().setMinSessions(0).build());
+    // Add a wait time for sessions to be initialized. In this case, since minSessions = 0, the
+    // wait time is for multiplexed sessions
+    builder.setSessionPoolOption(
+        SessionPoolOptions.newBuilder()
+            .setMinSessions(0)
+            .setWaitForMinSessions(Duration.ofSeconds(5))
```

Comment:
This test verifies the behavior of the client when a query is blocked while waiting for the session pool to have a session available. This changes that. Is there something in this test that does not work with multiplexed sessions if this wait time is not specified?

Reply:
Makes sense. I have now modified the code to 
```
if (sessionPoolOptions.getUseMultiplexedSession()) {
      sessionPoolOptions.toBuilder().setWaitForMinSessions(Duration.ofSeconds(5));
    }
```

The previous code actually had no impact on regular sessions test because minSessions = 0. And hence the blocking wait time did not actually block for the regular sessions.